### ISSUE TITLE
[HOLD for payment 2024-11-13] [HOLD for payment 2024-10-22] [$250] Ask our admin leads qualifying questions in Stage 1 and trigger "set up accounting integration" task

### ISSUE THREAD

**Issue by anmurali** [Expensify/App#48745]
P/S in slack [here](https://expensify.slack.com/archives/C07HPDRELLD/p1725057672522469?thread_ts=1725057659.236519&cid=C07HPDRELLD)

**Proposal**: Ask top down leads relevant qualifying questions so we can customize their onboarding tasks accordingly
**Problem**: Expensify caters to companies of all sizes, starting from sole props all the way to companies with 1000s of employees and several entities. Further the optimal setup of a workspace varies based on other conditions such as whether the company has an accounting integration or not. The way our current onboarding is designed, it assumes the same tasks, in the same order are important to all of them. This might not be a home run with every lead and might lead to a higher abandonment rate
**Hypothesis**: If we ask the lead a few simple qualifying questions and use those to show them tasks that we know other leads with similar answers prioritized before adding a card on file, then we have a higher chance of converting them to paid. So, if the same or more admins finish Stage 1 and give us this info, then having this info is more valuable than their company name and first/ last name
**Solution**:

1. In Stage 1 onboarding, for users that select `Manage my team’s expenses`, drop asking for Company name and User name. Instead ask for the number of employees they want to onboard and if they have an accounting package. [External]
![image](https://github.com/user-attachments/assets/742e2a4a-6c88-451b-9d38-e30691a6049f)

2. Customize the stage 2 onboarding tasks to include accounting integration if one is selected [Internal]
3. Show the accounting feature as enabled in workspace settings LHN if an accounting integration is selected [External]

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021833948133359014485</li>
        <li>Upwork Job ID: 1833948133359014485</li>
        <li>Last Price Increase: 2024-09-11</li>
<li>Automatic offers: </li>
<ul>
<li>nkdengineer | Contributor | 103936426</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @anmurali</details>

**Comment by dubielzyk-expensify** [comment]
Why are we dropping name on this one? Do we not still think that's useful? 

Mock without name selection:
![CleanShot 2024-09-09 at 16 40 34@2x](https://github.com/user-attachments/assets/65ef58c5-9983-4f77-a138-47839e4ab1d8)

**Comment by trjExpensify** [comment]
- >Don't worry you can change this later.

On this copy, can you actually change it later? If so, where? 

- >Expensify caters to companies of all sizes, starting from sole props all the way to companies with 1000s of employees and several entities.

I kinda' worry that our top end option being 50+ employees makes us seem like we're not a fit for even MM leads, let alone ENT leads that come looking, so I wonder if we tweak these a bit to include some larger company size ranges? I.e 

- 1-10
- 11-50
- 51-100
- 101-1000
- 1001+

Sidebar: given the qualifiers on the home page for 1-9 or 10+. Will we only show this screen to ask the question again if they've selected 10+ on the sign-up page? Seems a bit repetitive to ask it again for someone who has already told us they're a VSB with an employee count of 1-9. 

- Do you use any accounting software? 

Instead of "None" on the accounting solution list, should we change that to "None of the above" or something? For example, if I use Microsoft Dynamics, SAP, Oracle, Sage 50 etc as my accounting solution, "None" isn't an accurate choice for me to select to answer this question. I do use an accounting solution, just not one that has an out-of-the-box connection with Expensify.

**Comment by anmurali** [comment]
I think the intention behind saying "Don't worry you can change this later" was to convey the fact that this doesn't mean you can't add more employees. Maybe we can reword that to convey the point, which is - give us your best guess, you're not locked down to this range per se. cc @jamesdeanexpensify

**Comment by anmurali** [comment]
>I kinda' worry that our top end option being 50+ employees makes us seem like we're not a fit for even MM leads, let alone ENT leads that come looking, so I wonder if we tweak these a bit to include some larger company size ranges? I.e

I agree, let's use what @trjExpensify recommended for the ranges. 

>Instead of "None" on the accounting solution list, should we change that to "None of the above" or something? 

Also agreed, let's update that

@dubielzyk-expensify can you update the Figma but maybe let's wait for James to give us alternate copy for the size range sub text and we can update in one go?

**Comment by jamesdeanexpensify** [comment]
It feels like we don't even need to add that subtext at all - the question seems straightforward enough - "How many employees do you have?"

**Comment by marcaaron** [comment]
>Customize the stage 2 onboarding tasks to include accounting integration if one is selected

I think we need some copy for this. A **title** and **description**.

**Comment by jamesdeanexpensify** [comment]
Oh sorry, I was responding to my tag [here](https://github.com/Expensify/App/issues/48745#issuecomment-2339050464). Maybe @anmurali or @trjExpensify could confirm what's needed?

**Comment by marcaaron** [comment]
No worries, @jamesdeanexpensify! I was just generally noting that it was needed, but if you can help would be amazing 🙇

**Comment by dubielzyk-expensify** [comment]
![CleanShot 2024-09-10 at 12 24 50@2x](https://github.com/user-attachments/assets/65dd0fdc-4581-45e6-99ab-2d780a70502a)

Changed it to "More than 1000 employees" instead of "1001+ employees". Think it reads a bit better, but happy to change it.

Everything else is changed

**Comment by trjExpensify** [comment]
>Changed it to "More than 1000 employees" instead of "1001+ employees". Think it reads a bit better, but happy to change it.

Sounds good! 

>Oh sorry, I was responding to my tag https://github.com/Expensify/App/issues/48745#issuecomment-2339050464. Maybe @anmurali or @trjExpensify could confirm what's needed?

I don't think anything is **needed** per se, I'd also be down for starting with no sub-titles.

**Comment by jamesdeanexpensify** [comment]
@dubielzyk-expensify could you add a comma to "1000" so it's "1,000"? Thank you! I couldn't find the figma file. 

Am I needed for anything else in this issue? I'm happy to help, but I want to make sure what (if anything) remains!

**Comment by dubielzyk-expensify** [comment]
![CleanShot 2024-09-11 at 10 26 12@2x](https://github.com/user-attachments/assets/f8bbe5a3-bbac-4f68-afcf-bc1d95171941)

Done

**Comment by marcaaron** [comment]
>Customize the stage 2 onboarding tasks to include accounting integration if one is selected

Still need copy for this.

**Comment by anmurali** [comment]
I am waiting for Zach and Doza to come back from Netsuite conference to write this copy. But we're not blocked on building [this alternate flow ](https://github.com/Expensify/App/issues/48745#issuecomment-2342386517). Let's make that an External? @marcaaron

**Comment by dubielzyk-expensify** [comment]
Here's the mocks with More Fetures toggled on:

![CleanShot 2024-09-12 at 09 05 32@2x](https://github.com/user-attachments/assets/0b3d0cfe-fb58-41c4-a69c-a9b1773c04db)

**Comment by shawnborton** [comment]
Minor nitpicks but for the final screen there, does the intro text stay fixed and not scroll with the page? I would expect it to scroll with the page I think. Also, looks like we need to back the card padding down to 20px there too. Otherwise looking great!

**Comment by dubielzyk-expensify** [comment]
> Minor nitpicks but for the final screen there, does the intro text stay fixed and not scroll with the page? I would expect it to scroll with the page I think.

Agree that I also would expect this, but that's not happening in prod today. Let me file a bug. Screenshot of product:
![CleanShot 2024-09-12 at 14 35 49@2x](https://github.com/user-attachments/assets/8a92e5ca-334e-4986-8aaf-c37fd4b71405)

Fixed padding (and the intro text which I'll fix in a bug):
![CleanShot 2024-09-12 at 14 36 40@2x](https://github.com/user-attachments/assets/39009030-418c-438b-aa30-94aa714b9908)

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Ask our admin leads qualifying questions in Stage 1 and trigger "set up accounting integration" task

### What is the root cause of that problem?
This is a new feature

### What changes do you think we should make in order to solve the problem?

> In Stage 1 onboarding, for users that select Manage my team’s expenses, drop asking for Company name and User name. Instead ask for the number of employees they want to onboard and if they have an accounting package. [External]

1. We must create two new onboarding pages for the employee and accounting steps. The style of these pages can follow the existing pages. For the list on each page, we can use the `MenuItemList` or `SelectionList` to display. For the icon in the accounting step, we can follow the icon in `PolicyAccountingPage` and we need design team provide the none icon.

Then when we choose the `MANAGE_TEAM` here, we will navigate to the employee step and in employee step we will navigate to the accounting step

https://github.com/Expensify/App/blob/336e4782aec98bb29a06c28e0d630c276d4eddf6/src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx#L69-L72


2. Because we will drop the company name step then we should decide which step we will create the onboarding policy. I think we can create the onboarding policy when choosing the `MANAGE_TEAM` purpose here or we can create the new policy after the employee step as we do [here](https://github.com/Expensify/App/blob/336e4782aec98bb29a06c28e0d630c276d4eddf6/src/pages/OnboardingWork/BaseOnboardingWork.tsx#L41-L46)


https://github.com/Expensify/App/blob/336e4782aec98bb29a06c28e0d630c276d4eddf6/src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx#L69-L72

> Show the accounting feature as enabled in workspace settings LHN if an accounting integration is selected [External]

3. In the accounting step, if we select an accounting option, we will call `Policy.enablePolicyConnections` to enable the accounting feature or we can update the onyx data of the policy in `completeOnboarding` and BE will update the data in this API.
### What alternative solutions did you explore? (Optional)

**Comment by shawnborton** [comment]
Looks great @dubielzyk-expensify - and thanks for filing that bug!

**Comment by s77rt** [comment]
@nkdengineer Thanks for the proposal. Overall the suggested flow looks good to me.

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/48745#issuecomment-2345255157)

**Comment by marcaaron** [comment]
LGTM, but we are missing details about what to send to the API. I think we should start sending the `policyID`, `companySize`, and `userReportedIntegration` during the completeOnboarding call. So, you need some details about how we will achieve that.

**Comment by marcaaron** [comment]
BE is in progress on this. We can split it into two front end PRs.

1. Just add the flow and start sending the new data to the API
2. We will come back and add the task when calling `completeOnboarding()` once the BE change is deployed and copy is ready for that.

Sound good @nkdengineer ?

**Comment by nkdengineer** [comment]
@marcaaron That sounds good.

**Comment by nkdengineer** [comment]
@dubielzyk-expensify Does the close icon exist in the app?

**Comment by shawnborton** [comment]
I'm not sure if it does, so if you can't find it, here is a svg you can use: 
[close-circle.svg.zip](https://github.com/user-attachments/files/16992723/close-circle.svg.zip)

**Comment by dannymcclain** [comment]
>Does the close icon exist in the app?

It does! It's `App/assets/images/x-circle.svg`

**Comment by nkdengineer** [comment]
@s77rt The [PR](https://github.com/Expensify/App/pull/49161) is ready. Let's review two new pages first, and then we will integrate the API when the back-end change is complete.

**Comment by marcaaron** [comment]
Backend changes are on production now btw. We still need the copy for the tasks. @nkdengineer let's get this PR merged and then we will come back to add the relevant task data once we have it?

**Comment by davidcardoza** [comment]
I started working on the accounting task copy here https://docs.google.com/document/d/1AgQpxsUk7zcGJDVqlsMVXckD4c3LfDFa6Ppy7jj-xl0/edit

**Comment by marcaaron** [comment]
Great! We're almost ready here maybe we can actually get it done all in the same PR after all @nkdengineer.

There are a couple of small backend changes here we are blocked on:

- [ ] https://github.com/Expensify/Web-Expensify/pull/43605
- [ ] https://github.com/Expensify/Auth/pull/12542

**Comment by nkdengineer** [comment]
Got it.

**Comment by s77rt** [comment]
PR still under review. [We are almost there](https://github.com/Expensify/App/pull/49161#pullrequestreview-2328833178)

**Comment by marcaaron** [comment]
Backend changes are fully done now and front end is very close to done. I will be going OOO on Friday so it would be great if we can get the current frontend PR merged tomorrow if possible. And maybe we can create a new ticket for the "accounting task" itself.

**Comment by s77rt** [comment]
We still need translation and design confirmation https://github.com/Expensify/App/pull/49161#pullrequestreview-2328833178

cc @marcaaron @dubielzyk-expensify

**Comment by anmurali** [comment]
@s77rt @nkdengineer 

Re: The description in the employees step
Let's just remove the description from both steps because I think the title is self explanatory
![image](https://github.com/user-attachments/assets/11018ce7-6c1d-4190-b76f-97fc48b435d3)

**Comment by anmurali** [comment]
Translations are [here](https://docs.google.com/document/d/1D4ersWdRiLPWgLVQdVuOQYop2wOWgY9kMzU7tvFbBW8/edit#heading=h.nhtsmauf72jq)

**Comment by anmurali** [comment]
**The copy for the task we want to add**

> **Title**: **Connect to QuickBooks Online**
> 
> **Description**:  Connect to QuickBooks Online for automatic expense coding and syncing that makes month-end close a breeze. 
> 
> **Here’s how to connect to QuickBooks Online:**
> 
> 1. Click your profile photo.
> 2. Go to Workspaces.
> 3. Select your workspace.
> 4. Click Accounting.
> 5. Find QuickBooks Online.
> 6. Click Connect.
> 
> [Take me to Accounting!](link to Accounting)

**Comment by marcaaron** [comment]
@nkdengineer are we opening a second PR to add this task next? 

@anmurali To confirm, are we wanting to only add a task for QBO? Or a task for each integration? So far, we have not added any tasks after the user selects an integration.

**Comment by marcaaron** [comment]
@davidcardoza you might also know the answer if you're around.

**Comment by nkdengineer** [comment]
@marcaaron Sure.

**Comment by nkdengineer** [comment]
@marcaaron I will open the PR once we confirm the translation for all new tasks in mange my team's flow.

**Comment by nkdengineer** [comment]
@marcaaron Let's me know once the task copy is complete then I can create the second PR.

**Comment by davidcardoza** [comment]
> To confirm, are we wanting to only add a task for QBO? Or a task for each integration? So far, we have not added any tasks after the user selects an integration.

We will need to revise the copy per integration, that's correct. Do you need copy for each integration?

**Comment by marcaaron** [comment]
I'm guessing most of it is reusable and we'll just swap out the integration name, but not sure if `Connect to [integration] for automatic expense coding and syncing that makes month-end close a breeze.` makes sense for everything or if we want to promote anything else? What do you think?

**Comment by anmurali** [comment]
Yea, my bad, we need to add it for all integrations @marcaaron
Also, I don't think we are enabling `accounting` as a workspace feature for this workspace automatically when they select one of the integrations in Stage 1 yet. We need to do that too. 

> Title: Connect to QuickBooks Online
> 
> Description: **Connect to QuickBooks Online** for automatic expense coding and syncing that makes month-end close a breeze.
> 
> Here’s how to connect to QuickBooks Online:
> 
> 1. Click your profile photo.
> 2. Go to Workspaces.
> 3. Select your workspace.
> 4. Click **Accounting**.
> 5. Find **QuickBooks Online**.
> 6. Click **Connect**.
> 
> [Take me to Accounting!](link to Accounting)
> 


> Title: Connect to Xero
> 
> Description: **Connect to Xero** for automatic expense coding and syncing that makes month-end close a breeze.
> 
> Here’s how to connect to Xero:
> 
> 1. Click your profile photo.
> 2. Go to Workspaces.
> 3. Select your workspace.
> 4. Click **Accounting**.
> 5. Find **Xero**.
> 6. Click **Connect**.
> 
> [Take me to Accounting!](link to Accounting)
> 

> Title: Connect to NetSuite
> 
> Description: **Connect to NetSuite** for automatic expense coding and syncing that makes month-end close a breeze.
> 
> Here’s how to connect to NetSuite:
> 
> 1. Click your profile photo.
> 2. Go to Workspaces.
> 3. Select your workspace.
> 4. Click **Accounting**.
> 5. Find **NetSuite**.
> 6. Click **Connect**.
> 
> [Take me to Accounting!](link to Accounting)
> 

> Title: Connect to Sage Intacct
> 
> Description: **Connect to Sage Intacct** for automatic expense coding and syncing that makes month-end close a breeze.
> 
> Here’s how to connect to Sage Intacct:
> 
> 1. Click your profile photo.
> 2. Go to Workspaces.
> 3. Select your workspace.
> 4. Click **Accounting**.
> 5. Find **Sage Intacct**.
> 6. Click **Connect**.
> 
> [Take me to Accounting!](link to Accounting)
>

**Comment by marcaaron** [comment]
>Also, I don't think we are enabling accounting as a workspace feature for this workspace automatically when they select one of the integrations in Stage 1 yet. We need to do that too.

Oh, yes, there is a bug in the original PR where we are not sending the policyID to the API. @nkdengineer is working on the fix I believe.

**Comment by nkdengineer** [comment]
@anmurali Do I need to confirm the translation for copied tasks above or can you provide the translation for this, thanks.

**Comment by marcaaron** [comment]
We only need to translate one and then just sub out the name (if my eyes are working 😄). I will ask someone.

**Comment by jamesdeanexpensify** [comment]
You can use [this GPT](https://chatgpt.com/g/g-2dgOQl5VM-english-to-spanish-translator-aka-jaimegpt) for initial translation and then get human verification in #espanol-translations!

**Comment by marcaaron** [comment]
Done [here](https://expensify.slack.com/archives/C21FRDWCV/p1729028779025959). I don't think contributors have access to that channel so I will post the correct translation when someone responds.

**Comment by marcaaron** [comment]
> **Title:** Conectar a QuickBooks Online
> **Description:** Conéctate a QuickBooks Online para la clasificación y sincronización automática de gastos, lo que facilita el cierre de fin de mes.
> 
> Aquí te mostramos cómo conectarte a QuickBooks Online:
> Haz clic en tu foto de perfil.
> Ve a Espacios de trabajo.
> Selecciona tu espacio de trabajo.
> Haz clic en Contabilidad.
> Encuentra QuickBooks Online.
> Haz clic en Conectar. [¡Llévame a Contabilidad!](enlace a Contabilidad)

@nkdengineer for the rest, we can just swap out the names in the relevant places.

**Comment by nkdengineer** [comment]
@marcaaron Just have one more question here, we will keep all existing tasks for `onboardingChoices.MANAGE_TEAM` and only add the accounting task, right

**Comment by marcaaron** [comment]
Correct, no tasks are being removed only added.

**Comment by nkdengineer** [comment]
@marcaaron What is the type of the new task? I added it as `integration` and back-end returns error.

<img width="1512" alt="Screenshot 2024-10-21 at 17 57 55" src="https://github.com/user-attachments/assets/98fb1c51-a523-44d0-9997-3e5295ca347c">

**Comment by marcaaron** [comment]
`addAccountingIntegration`

**Comment by marcaaron** [comment]
Final PR is in progress.

**Comment by anmurali** [comment]
> Also, I don't think we are enabling accounting as a workspace feature for this workspace automatically when they select one of the integrations in Stage 1 yet. We need to do that too.

Did we also do this?

**Comment by nkdengineer** [comment]
@s77rt The second PR is [here](https://github.com/Expensify/App/pull/51070).

> Did we also do this?

@anmurali  I did this and it works well now

**Comment by marcaaron** [comment]
Second PR is wrapped here and just waiting for that to go to prod.

And we are tracking the most recent change related to this feature [here](https://github.com/Expensify/App/issues/51591).

**Comment by anmurali** [comment]
Got @trjExpensify assigned in case we need to process payments while I am OOO

**Comment by trjExpensify** [comment]
Confirming it's just a pay out for the one PR here? https://github.com/Expensify/App/issues/48745#issuecomment-2458439720

**Comment by s77rt** [comment]
It's two PRs but one payment since the work was split https://github.com/Expensify/App/pull/49161 and https://github.com/Expensify/App/pull/51070

**Comment by trjExpensify** [comment]
>It's two PRs but one payment since the work was split https://github.com/Expensify/App/pull/49161 and https://github.com/Expensify/App/pull/51070

Gotcha, so by "one payment" you mean it's $250 total still?

**Comment by s77rt** [comment]
Yes but since there was a regression https://github.com/Expensify/App/issues/50637 it's $125 now

**Comment by trjExpensify** [comment]
Okay, perfect. Thanks for confirming! 

Payment summary as follows: 

- $125 to @s77rt for the C+ review (go ahead and request!) 
- $125 to @nkdengineer for the fix (paid!)

I checked [this regression test](https://expensify.testrail.io/index.php?/cases/view/2990420&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=229043) for the manageMyTeam flow, and it reflects employees and accounting. So I don't think we need an edit there. 

All settled up, closing!

**Comment by JmillsExpensify** [comment]
$125 approved for @s77rt
